packages/generator-nitro/generators/app/templates/src/shared/utils/typo/css/typo.scss

```diff
@@ -1,178 +1,209 @@
+	}
+	@if (map-has-key($font-size-definition, 'text-transform')) {
+		text-transform: map-get($font-size-definition, 'text-transform');
 	}
-	@return $typography-font-definition;
 }
 
-@mixin getDefinition($font-definition, $style, $family) {
-	font-size: map-get($font-definition, 'font-size');
-	line-height: map-get($font-definition, 'line-height');
-	letter-spacing: map-get($font-definition, 'letter-spacing');
-
-	@if($family == 'sans') {
-		font-family: Arial, sans-serif;
-
-		@if ($style == 'bold') {
-			font-family: Arial, sans-serif;
-			font-weight: bold;
-		}
-		@if ($style == 'italic') {
-			font-family: Arial, sans-serif;
-			font-style: italic;
-		}
+// Return the font definition for the given viewport and font-name
+@function get-font-definition($font-bundle, $viewport) {
+	@if (not(map-has-key($typography-size-definition, $font-bundle))) {
+		@error 'Typography font name `#{$font-bundle}` does not exist.';
+	} @else if (not(map-has-key(map-get($typography-size-definition, $font-bundle), $viewport))) {
+		@error 'Typography viewport size `#{$viewport}` does not exist for font `#{$font-bundle}`.';
 	}
-	@if($family == 'serif') {
-		font-family: Arial, sans-serif;
 
-		@if ($style == 'serif') {
-			font-family: Arial, sans-serif;
-		}
-		@if ($style == 'semi-bold') {
-			font-family: Arial, sans-serif;
-		}
+	$all-font-definition: ();
+	@if (map-has-key(map-get($typography-size-definition, $font-bundle), 'all')) {
+		$all-font-definition: map-get(map-get($typography-size-definition, $font-bundle), 'all');
 	}
-}
-
-/*
-	mixin to use in the code
-
-	Params:
-		- $size: gets the typo definition according to the size (e.g. s, m, l)
-		- $style: defines the style of the font. Accepted Values are 'bold', 'italic', 'semi-bold'. Default is null
-		- $family: defines the family of the language. Currently is minion default.
-		- $margin-bottom: adds a calculated margin at the bottom.
-*/
+	$font-definition: map-merge($all-font-definition, map-get(map-get($typography-size-definition, $font-bundle), $viewport));
 
-@mixin typography($size: $typography-default-size, $style: $typography-default-style, $family: $typography-default-family, $margin-bottom: 0) {
+	@return $font-definition;
+}
 
-	$font-definition: typography-get-font-definition('mobile', $size, $family);
-	@include getDefinition($font-definition, $style, $family);
+// Returns a (responsive) typography for the given font name
+// Usage:
+// span { @include typography('body') }
+// span { @include typography('body', 'md') }
+@mixin typography($font-bundle, $viewport: 'all') {
+	@if ($viewport == 'all') {
+		@include _generate-font-size(get-font-definition($font-bundle: $font-bundle, $viewport: 'mobile'));
 
-	@media screen and (min-width: $size-md-min) {
-		$font-definition: typography-get-font-definition('desktop', $size, $family);
-		@include getDefinition($font-definition, $style, $family);
+		@media screen and (min-width: $size-md-min) {
```

Comment:
Does screen make sense here, or can we save the bytes?

Reply:
I removed it..as we usually use a media-query mixin, that will replace that line anyway :-)